src/dbus_api/api/fetch_properties_2_4/methods.rs

```diff
@@ -0,0 +1,99 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+use std::collections::HashMap;
+
+use dbus::{
+    arg::{RefArg, Variant},
+    tree::{MTFn, MethodInfo, MethodResult},
+    Message,
+};
+use itertools::Itertools;
+
+use crate::dbus_api::{
+    api::shared::{list_keys, locked_pool_uuids, locked_pools},
+    consts,
+    types::TData,
+    util::result_to_tuple,
+};
+
+const ALL_PROPERTIES: [&str; 4] = [
+    consts::KEY_LIST_PROP,
+    consts::LOCKED_POOLS,
+    consts::LOCKED_POOL_UUIDS,
+    consts::LOCKED_POOL_DEVS,
+];
+
+pub fn locked_pools_with_devs(
+    info: &MethodInfo<MTFn<TData>, TData>,
+) -> Result<HashMap<String, Variant<Box<dyn RefArg>>>, String> {
```

Comment:
I don't understand exactly why the value type of the HashMap is a Variant, instead of being a HashMap of String * Variant.

Reply:
Resolved.